platform_channels/lib/src/platform_message_channel.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+
+/// This class manages three [BasicMessageChannel]s which uses [StringCodec],
+/// [JSONMessageCodec], and [BinaryCodec] respectively to send the message to
+/// platform and receive a reply.
+class PlatformMessages {
+  static final _stringCodecChannel =
+      BasicMessageChannel('stringCodecDemo', StringCodec());
+
+  static final _jsonMessageCodecChannel =
+      BasicMessageChannel<dynamic>('jsonMessageCodecDemo', JSONMessageCodec());
```

Comment:
Why is this channel the only one with an explicitly dynamic type? Are the others inferred as different types, or are they implicitly dynamic?

Reply:
The others have different types, for the channel using `BinaryCodec` it return `ByteData` and for `StringCodec` it will return `String`. The `JSONMessageCodec` is implicitly dynamic but implicit-dynamic is set to false in the analyzer, due to which I have to provide explicit type.